es_ES/design-principles.md

```diff
@@ -0,0 +1,243 @@
+
+* No deben existir API que fomenten las malas prácticas. e.g. no
+  `document.write()`, `innerHTML`, `insertAdjacentHTML()`, etc.
+
+  - La manipulación de cadenas para generar datos o código que posteriormente serán interpretados o analizados es una mala práctica, ya que conduce a vulnerabilidades de inyección de código.
+
+  - Si una operación es cara, ese gasto debe ser representado en la API (por ejemplo, devolviendo un `Future` o un `Stream`).  Evite proporcionar APIs que oculten el gasto de las tareas.
+
+* Las APIs de conveniencia que envuelven algún aspecto de un servicio de un entorno para su exposición en otro entorno (por ejemplo, exponer una API Android en Dart), deberían exponer/envolver la API completa, de modo que no haya obstáculos cognoscitivos a la hora de interactuar con ese servicio (en el que está bien usar la API expuesta hasta cierto punto, pero más allá de eso tiene que aprender todo sobre el servicio subyacente).
+
+* Las APIs que envuelven los servicios subyacentes pero impiden el acceso directo a la API subyacente (por ejemplo, cómo `dart:ui` expone a Skia) deben exponer cuidadosamente sólo las mejores partes de la API subyacente. Esto puede requerir características de refactorización para que sean más utilizables. Puede significar evitar exponer características de conveniencia que se abstraen sobre operaciones costosas a menos que haya una clara ganancia de rendimiento al hacerlo. Una superficie API más pequeña es más fácil de entender.
+
+  - Es por eso que `dart:ui` no expone `Path.fromSVG()`: lo comprobamos, y es igual de rápido hacer ese trabajo directamente en Dart, así que no hay beneficio en exponerlo. De esta manera, evitamos los costes (las estructuras de API más grandes son más caras de mantener, documentar y probar, y suponen una carga de compatibilidad para la API subyacente).
+
+
+Bugs
+----
+
+Asigne un error a sí mismo sólo cuando esté trabajando activamente en él. Si no estás trabajando en ello, déjalo sin asignar. No asigne bugs a personas a menos que sepa que ellos van a trabajar en ello. Si se encuentra con errores asignados en los que no va a trabajar en un futuro muy cercano, desasigne el error para que otras personas se sientan capacitadas para trabajar en ellos.
+
+Se puede escuchar a los miembros del equipo referirse a "lamer la galleta". Asignarse un error a sí mismo, o indicar que se va a trabajar en él, les dice a otros miembros del equipo que no lo solucionen. Si entonces no trabajas en ello inmediatamente, estás actuando como alguien que ha tomado una galleta, la ha lamido para no atraer a otras personas, y luego no se la ha comido de inmediato. Por extensión, "desclasificar la cookie" significa indicar al resto del equipo que no vas a trabajar en el error. inmediatamente después de todo, por ejemplo, desasignando el error de usted mismo.
```

Comment:
Por extensión, “unlicking the cookie” (deslamer la galleta) significa indicar al resto del equipo que no vas a trabajar en el error. inmediatamente después de todo, por ejemplo, desasignándote el error a ti mismo.